Palaso.MSBuildTasks/UnixName.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2014 SIL International
+// This software is licensed under the MIT License (http://opensource.org/licenses/MIT)
+// Parts based on code by MJ Hutchinson http://mjhutchinson.com/journal/2010/01/25/integrating_gtk_application_mac
+
+using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Palaso.BuildTasks
+{
+	public class UnixName : Task
+	{
+		public override bool Execute()
+		{
+			if (Environment.OSVersion.Platform == PlatformID.Unix)
+			{
+				IntPtr buf = IntPtr.Zero;
+				try
+				{
+					buf = System.Runtime.InteropServices.Marshal.AllocHGlobal(8192);
+					// This is a hacktastic way of getting sysname from uname ()
+					if (uname(buf) == 0)
+						Value = System.Runtime.InteropServices.Marshal.PtrToStringAnsi(buf);
```

Comment:
I assume uname() NUL-terminates each of the fields in the output struct.  If so, this looks like a suitable hack to obtain the value of the first field (sysname).